lib/camunda-platform/features/modeling/behavior/DeleteRetryTimeCycleBehavior.js

```diff
@@ -0,0 +1,126 @@
+import inherits from 'inherits';
+
+import {
+  getBusinessObject,
+  is
+} from 'bpmn-js/lib/util/ModelUtil';
+
+import {
+  find
+} from 'min-dash';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+const HIGH_PRIORITY = 15000;
+
+
+/**
+ * Camunda BPMN specific `camunda:FailedJobRetryTimeCycle` behavior.
+ *
+ * When `camunda:asyncAfter` or `camunda:asyncBefore` are set to false
+ * on an element, then `camunda:FailedJobRetryTimeCycle` shall be removed. This ensures
+ * that the BPMN diagram XML reflects the behavior of Camunda Platform engine.
+ */
+export default function DeleteRetryTimeCycleBehavior(eventBus) {
+
+  CommandInterceptor.call(this, eventBus);
+
+  this.executed([ 'properties-panel.update-businessobject', 'element.updateProperties' ],
+    HIGH_PRIORITY, function(context) {
+      const {
+        element,
+        properties
+      } = context;
+
+      const businessObject = getBusinessObject(element),
+            extensionElements = businessObject.extensionElements;
+
+      // (1.1) Execute if...
+      if (is(element, 'camunda:AsyncCapable') && // ...asyncCapable
+        properties && // ...properties updated
+        (properties['camunda:asyncBefore'] === false || // ...update async (1)
+          properties['camunda:asyncAfter'] === false) && // ...update async (2)
+        (extensionElements && extensionElements.values.length) && // ...we have extensionElements
+        (extensionElements.values.find(ele => is(ele, 'camunda:FailedJobRetryTimeCycle'))) // ...we have retryTimeCycle
+      ) {
+
+        // (1.2) ... but don't execute if one async stays or a TimerEventDefinition exists
+        if (isAsyncBefore(businessObject) || isAsyncAfter(businessObject) ||
```

Comment:
That's the same we have in https://github.com/camunda/camunda-bpmn-js/pull/31#discussion_r675497622 right?

Reply:
No, here it is different because we hook into the command after it was executed.

Reply:
I see, thanks for clarifying 👍

Reply:
Added a specific test for it: https://github.com/camunda/camunda-bpmn-js/pull/33/commits/2876e1b508090685379d3d61dcd00b63d3768618